### PR TITLE
Add a depth camera example

### PR DESCRIPTION
**Pull request by srmainwaring**
# 🎉 New feature

This PR adds an example that demonstrates how to use the `DepthCamera`.

## Summary

This is a standalone example that demonstrates how to render a scene using a depth camera. It runs using either the `ogre` or `ogre2` render engines.

depth_camera_ogre

The code for converting the depth camera image to a RGB grayscale format is adapted from `ign-sensors/src/DepthCameraSensor.cc`.

## Test it

The tutorial explains how to build and run the example in detail. Once built the example is run from the command line:

```bash
# run with the default render engine (ogre2)
./depth_camera

# run with the ogre render engine
./depth_camera ogre

# run with the ogre2 render engine and metal render system
./depth_camera ogre2 metal
```

The example accepts an additional command line argument to specify the render system. This will only take effect once https://github.com/ignitionrobotics/ign-rendering/pull/463 is merged. The fallback behaviour is to use the default render system GL3+.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: tutorials/23_depth_camera_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+registers a callback with the camera that captures a copy of
+the depth image each frame. This will later be drawn to the window in
+the main update loop.
+
+\snippet examples/depth_camera/GlutWindow.cc init camera
+
+The callback function `OnNewDepthFrame()` converts the depth image
+to a RGB grayscale format and copies it from the depth camera to a
+memory reserved in a globally scoped `Image`.
+The reason for setting the camera image format to `PixelFormat::PF_FLOAT32_RGBA`
```
[the remark this answers is not in the document]
Comment (srmainwaring) [reply tutorials/23_depth_camera_tutorial.md:58]: Good suggestion. I'll update the doc with your changes.

**Comment by srmainwaring** [comment]
> View control does not work well with a depth camera though.

I'll take a quick look at this. I copied the mouse handling from the segmentation camera example, but realise this has some comments about it effectiveness. I think the mouse handling in the ogre2demo works well, so I'll look into updating this example based on that.

**Comment by srmainwaring** [comment]
> View control does not work well with a depth camera though. This is expected as we also ran into this problem with the segmentation camera demo. I think this can be fixed later.

Here's a possible approach to fixing the view control for the depth, segmentation and thermal cameras: https://github.com/srmainwaring/ign-rendering/commit/2972a98ab7f87d1df5c5e9daa418ae1dfaf6df39

**Review by iche033 — APPROVED** [review]
looks good to me :)

**Pull request merged**